### PR TITLE
Init e2e tests 

### PR DESCRIPTION
**Pull request by hiaux0**
## What was done
1. Add some tests for landing page and Deals info
2. (copy from [`cypress/readme.md`](https://github.com/PrimeDAO/prime-deals-dapp/tree/internal/init-e2e-tests/cypress)):

  ## Concepts:
  - [Cypress](https://docs.cypress.io/guides/overview/why-cypress)
    - [Features](https://docs.cypress.io/guides/overview/key-differences#Architecture)
  - [BDD](https://cucumber.io/docs/bdd/)
    - [Gherkin][gherkin]
    - [Cucumber](https://cucumber.io/docs/cucumber/?sbsearch=Cucumber) - Implementation of BDD in code
      - [cypress-cucumber-preprocessor](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor)
        - [the fork we are using](https://github.com/badeball/cypress-cucumber-preprocessor)

  ## Setup
  ```
  + --- cypress/
  |     + --- fixtures/
  |     + --- integration/
  |           + --- features/    // (1) BDD part
  |           + --- tests/       // (2) Cypress part
  |     + --- plugins/
  |     + --- support/
  ```

  ### (1) BDD
  Check out few thoughts on BDD in [Notion](https://www.notion.so/primedao/Why-BDD-100ad3b686054579a2f1c558e8646633)

  ### (2) Cypress
  Check out few thoughts on Cypress in [Notion](https://www.notion.so/primedao/E2E-aed92af51bf74634b0d89027fe0d7817)

  ## Development

  ### Commands
  `npm run e2e`
  - Open a Cypress GUI, where you can develop interactively
    - Note the "Cypress Studio" feature to generate tests based on your interactions

  ### Flow
  1. Single out a Specification, that you want to cover with automated tests
  2. Define the Specifications in [Gherkin][gherkin] format in `.feature` files
  3. Write corresponding Cypress test with file convention `.e2e.ts`

  [gherkin]: https://cucumber.io/docs/gherkin/

### CODE REVIEW

File: src/services/EthereumService.ts [hunk 1]
```diff
@@ -136,6 +138,8 @@ export class EthereumService {
       throw new Error(`Please connect to either ${Networks.Mainnet} or ${Networks.Rinkeby}`);
     }
 
+    if (isCypress) return;
+
     // comment out to run DISCONNECTED
```
Comment (hiaux0) [inline src/services/EthereumService.ts:143]: @dkent600 

I wanted to run the tests disconnected, to reduce noise from outside.
Let me know what you think, as this can be easily reverted.

File: cypress/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# *WIP*
+
+## Concepts:
```
Comment (dkent600) [inline cypress/README.md:3]: can we describe methods and features, etc?

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: src/services/EthereumService.ts (hunk 1 above)
Comment (hiaux0) [reply src/services/EthereumService.ts:143]: > the ability to run connected?

Clarifying my intention.
I wanted to reduce the amount of connections, until a test is implemented for a particular connection.
Ie. it is unclear to me (lack of knowledge about Deals and web3), how the "connections" affect the app.
So I wanted to write tests until I encounter a problem, where I _need_ to enable the connections again.

Anyway, I'll remove this, as your points are also valid.

File: cypress/README.md (hunk 2 above)
Comment (hiaux0) [reply cypress/README.md:3]: I'm a bit hesitant on this one.
How far should I go into explaining BDD and Cypress.
As I see it, it's better to spend time to look and the official docs, which I linked, and consult our tests for examples.
If I add descriptions here, it's just duplication imo, which also get outdated when the underlying frameworks change.

File: cypress/integration/common/navigate.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+
```
[the remark this answers is not in the document]
Comment (hiaux0) [reply cypress/integration/common/navigate.ts:1]: I'll move one of the Steps into /deals, and will keep the `visit("/")` one in this common place.
Former is specific to deals, latter is "common" enough imo.

File: cypress/README.md (hunk 2 above)
Comment (dkent600) [reply cypress/README.md:3]: I'm only looking for very basic descriptions.

File: cypress/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# *WIP*
```
[the remark this answers is not in the document]
Comment (hiaux0) [reply cypress/README.md:1]: I addressed the first question by adding a README.md to /test root.
With a small note on differences.

File: test/README.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# Unit tess
+
+# E2E (Cypress)
+[Cypress Readme](/cypress/README.md)
+
+## Difference of unit testing and E2E
+- Unit: Test one part of a component/code in isolation (--> "a unit")
+  - We are using Jest (alternatives: Jasmine, Mocha)
+- E2E: Test many components/code interacting (--> End to End)
+  - We are using Cypress (alternatives: Playwright, others)
```
Comment (hiaux0) [inline test/README.md:14]: [In this comment](https://github.com/PrimeDAO/prime-deals-dapp/pull/63#discussion_r780703686) you mentioned
> Put this in the context of other tests that don't require a UI?

Can you rephrase? Is this explanation here enough?

File: cypress/README.md (hunk 2 above)
Comment (dkent600) [reply cypress/README.md:3]: I'm only looking for very basic descriptions.  Can go in the top level test readme.

**Review by dkent600 — CHANGES_REQUESTED** [review]

File: cypress/README.md (hunk 2 above)
Comment (hiaux0) [reply cypress/README.md:3]: I thought I did this via 65e1f54.
What is still missing?

Comment (dkent600) [reply cypress/README.md:3]: Looks specifically for a quick summary about "methods" and "features".  A quick enumeration of the different file types (extensions) would be helpful too.  We need a quick sense of the steps involved in creating a test.  Nothing detailed, just very high level.

Comment (hiaux0) [reply cypress/README.md:3]: > top level test readme

I hope we both understand /cypress for this piece of documentation

> high level

Done via 9fa56a4

**Review by dkent600 — APPROVED** [review]
